src/app/components/EditorInterface.tsx

```diff
@@ -101,25 +118,54 @@ const EditorInterface = () => {
   useEffect(() => {
     if (appState?.appState.readOnly) {
       const template = Handlebars?.compile(originalTemplate || '')
-      const mockData = appState.appState.mockData.filter(
-        (el) => el.givenName === appState.appState.selectedClient,
-      )[0]
-      const c = template({ client: mockData })
-      editor?.chain().focus().setContent(c).run()
+      const _client = appState.appState.clientList.find(
+        (el) => el.id === (appState.appState.selectedClient as IClient).id,
+      )
+      const client = {
+        ..._client,
+        company: appState?.appState.selectedClientCompanyName,
+      }
+      const c = template({ client: client })
+      setTimeout(() => {
+        editor?.chain().focus().setContent(c).run()
+      })
     } else {
-      editor
-        ?.chain()
-        .focus()
-        .setContent(originalTemplate as string)
-        .run()
+      setTimeout(() => {
+        editor
+          ?.chain()
+          .focus()
+          .setContent(originalTemplate as string)
+          .run()
+      })
     }
-  }, [appState?.appState.selectedClient])
+  }, [
+    appState?.appState.selectedClient,
+    appState?.appState.selectedClientCompanyName,
+  ])
 
   useEffect(() => {
     if (appState?.appState.readOnly) return
     setOriginalTemplate(editor?.getHTML())
   }, [editor?.getText(), appState?.appState.readOnly])
 
+  useEffect(() => {
+    if (!appState?.appState.settings) return
+    if (
+      originalTemplate !== appState?.appState.settings.content ||
+      appState?.appState.settings.backgroundColor !==
+        appState?.appState.editorColor
+    ) {
+      appState?.toggleChangesCreated(true)
+    } else {
+      appState?.toggleChangesCreated(false)
+    }
+  }, [
+    originalTemplate,
+    appState?.appState.editorColor,
+    appState?.appState.bannerImg,
+    appState?.appState.readOnly,
+  ])
+
   useEffect(() => {
     if (!editor) return
```

Comment:
let's not return undefined from useEffect since react will treat it as clean up function and try to invoke the function during component unmount